modules/core/src/main/scala/doobie/util/transactor.scala

```diff
@@ -186,6 +188,19 @@ object transactor {
           }.scope
       }
 
+    def transBuffer(bufferSize: Int)(implicit ev: Concurrent[M]): Stream[ConnectionIO, *] ~> Stream[M, *] =
+      new (Stream[ConnectionIO, *] ~> Stream[M, *]) {
+        def apply[T](s: Stream[ConnectionIO, T]) = {
+          fs2.Stream.eval(Queue.bounded[M, Option[T]](bufferSize)).flatMap { buffer =>
+            val res = Stream.resource(connect(kernel)).flatMap { c =>
+              Stream.resource(strategy.resource).flatMap(_ => s).translate(run(c))
+                .evalMap(x => buffer.offer(Some(x))) ++ Stream.emit(None).evalMap(buffer.offer)
+            }
+            Stream.fromQueueNoneTerminated(buffer).concurrently(res)
+          }
+        }
+      }
```

Comment:
Perhaps we don't need to add a new method here, since users will most likely be calling `someStream.transact`

Reply:
Yeah agree. Where do you think is the best place to put the `bufferSize` config?

Reply:
Maybe something like this in `StreamOps`?

```
  def transactPrefetchN[M[_]: MonadCancelThrow: Concurrent](xa: Transactor[M], numChunks: Int)(implicit
      ev: Stream[F, A] =:= Stream[ConnectionIO, A],
  ): Stream[M, A] = xa.transP.prefetchN(numChunks)
```